service/matching/tasklist/task_list_manager.go

```diff
@@ -268,49 +286,112 @@ func (c *taskListManagerImpl) handleErr(err error) error {
 	return err
 }
 
-func (c *taskListManagerImpl) loadTaskListPartitionConfig() {
-	if c.taskListID.IsRoot() {
-		return
-	}
+func (c *taskListManagerImpl) TaskListPartitionConfig() *types.TaskListPartitionConfig {
 	c.partitionConfigLock.RLock()
 	if c.partitionConfig != nil {
+		config := *c.partitionConfig
 		c.partitionConfigLock.RUnlock()
-		return
+		c.logger.Debug("get task list partition config from db", tag.Dynamic("root-partition", c.taskListID.GetRoot()), tag.Dynamic("config", config))
+		return &config
 	}
 	c.partitionConfigLock.RUnlock()
+	return nil
+}
 
-	c.partitionConfigLock.Lock()
-	if c.partitionConfig != nil {
+func isTaskListPartitionConfigEqual(a types.TaskListPartitionConfig, b types.TaskListPartitionConfig) bool {
```

Comment:
this function is not future proof and easy to miss while changing the `TaskListPartitionConfig`. Let's do comparison using reflect or cmp